standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1062,6 +1062,23 @@ public void createTable(Table tbl) throws InvalidObjectException, MetaException
         rollbackTransaction();
       }
     }
+
+    // Update column statistics if available.
+    if (tbl.isSetColStats()) {
+      try {
+        ColumnStatistics colStats = tbl.getColStats();
+        // Set the normalized names directly instead of normalising those again
+        colStats.getStatsDesc().setCatName(mtbl.getDatabase().getCatalogName());
+        colStats.getStatsDesc().setDbName(mtbl.getDatabase().getName());
+        colStats.getStatsDesc().setTableName(mtbl.getTableName());
+        updateTableColumnStatistics(colStats, null, tbl.getWriteId());
```

Comment:
Stats replication for ACID tables will be tackled in a separate task https://issues.apache.org/jira/browse/HIVE-21109. For non-ACID tables passing null writeId list and an invalid writeId suffices since they don't have any writeId associated with them.

Reply:
sounds good.